api/app/signals/apps/signals/tests/management/commands/delete_signals/test_delete_signals.py

```diff
@@ -0,0 +1,121 @@
+        self.assertTrue(Signal.objects.filter(id=signal.id).exists())
+        self.assertFalse(DeletedSignal.objects.filter(signal_id=signal.id).exists())
+
+    def test_signals_should_not_be_deleted(self):
+        """
+        Signal created 6 days ago and the final state set 5 days ago. So the signals should not be deleted.
+        """
+        with freeze_time(timezone.now() - timezone.timedelta(days=6)):
+            signal = SignalFactory.create(status__state=AFGEHANDELD)
+
+        buffer = StringIO()
+        call_command('delete_signals', f'{AFGEHANDELD}', '365', stdout=buffer)
+        output = buffer.getvalue()
+
+        self.assertNotIn(f'Deleted Signal: #{signal.id}', output)
+        self.assertTrue(Signal.objects.filter(id=signal.id).exists())
+        self.assertFalse(DeletedSignal.objects.filter(signal_id=signal.id).exists())
+
+    def test_signals_should_be_deleted(self):
+        """
+        Signal created 370 days ago and the final state set 369 days ago. So the signals should be deleted.
+        """
+        with freeze_time(timezone.now() - timezone.timedelta(days=370)):
+            signal = SignalFactory.create(status__state=GEANNULEERD)
+
+        buffer = StringIO()
+        call_command('delete_signals', f'{GEANNULEERD}', '365', stdout=buffer)
+        output = buffer.getvalue()
+
+        self.assertIn(f'Deleted Signal: #{signal.id}', output)
+        self.assertFalse(Signal.objects.filter(id=signal.id).exists())
+        self.assertTrue(DeletedSignal.objects.filter(signal_id=signal.id).exists())
+
+    def test_signals_should_not_be_deleted_wrong_states(self):
+        """
+        Signal created 370 days ago and the wrong state set 369 days ago. So the signals should be deleted.
+        """
+        with freeze_time(timezone.now() - timezone.timedelta(days=370)):
+            signal = SignalFactory.create(status__state=GEMELD)
+
+        buffer = StringIO()
+        call_command('delete_signals', f'{AFGEHANDELD}', '365', stdout=buffer)
+        output = buffer.getvalue()
+
+        self.assertNotIn(f'Deleted Signal: #{signal.id}', output)
+        self.assertTrue(Signal.objects.filter(id=signal.id).exists())
+        self.assertFalse(DeletedSignal.objects.filter(signal_id=signal.id).exists())
+
+    def test_parent_and_child_signals_should_be_deleted(self):
+        """
+        Parent signal created 370 days ago and the final state set 369 days ago. So the signal and its children should
+        be deleted.
+        """
+        with freeze_time(timezone.now() - timezone.timedelta(days=370)):
+            parent_signal = SignalFactory.create(status__state=GEANNULEERD)
+            child_signal = SignalFactory.create(parent=parent_signal)
+
+        buffer = StringIO()
+        call_command('delete_signals', f'{GEANNULEERD}', '365', stdout=buffer)
+        output = buffer.getvalue()
+
+        self.assertIn(f'Deleted Signal: #{parent_signal.id}', output)
```

Comment:
Are you relying on the fact that in the local scope the `parent_signal` is still available although behind the scenes it is already deleted? Is it cleaner to copy the ID of the `parent_signal` to `parent_signal_id` and not rely on the behavior above?

Same for `child_signal` below.

Reply:
The state of the signal here is still the same, this will only change when executing `refresh_from_db`